src/main/java/io/confluent/connect/jdbc/source/JdbcSourceConnectorConfig.java

```diff
@@ -252,6 +258,31 @@ public boolean visible(String name, Map<String, Object> config) {
     }
   }
 
+  /**
+   * A recommender for table names that caches the table names for the last configuration.
+   */
+  private static class CachingTableRecommender extends TableRecommender {
+
+    private Map<String, Object> lastConfig;
+    private List<Object> results;
+
+    @Override
+    public List<Object> validValues(String name, Map<String, Object> config) {
+      if (lastConfig != null && lastConfig.equals(config)) {
```

Comment:
The connector returns a static instance https://github.com/confluentinc/kafka-connect-jdbc/blob/master/src/main/java/io/confluent/connect/jdbc/JdbcSourceConnector.java#L156 https://github.com/rhauch/kafka-connect-jdbc/blob/a8dfbdb43794652dc490dd769ee68d6ddb93365e/src/main/java/io/confluent/connect/jdbc/source/JdbcSourceConnectorConfig.java#L224 so the same recommender will always be used, no?